x-pack/plugins/ml/public/application/jobs/new_job/pages/components/common/json_editor_flyout/json_editor_flyout.tsx

```diff
@@ -110,19 +145,51 @@ export const JsonEditorFlyout: FC<Props> = ({ isDisabled, jobEditorMode, datafee
                     defaultMessage: 'Job configuration JSON',
                   })}
                   value={jobConfigString}
+                  heightOffset={showChangedIndicesWarning ? WARNING_CALLOUT_OFFSET : 0}
                 />
               )}
               {datafeedEditorMode !== EDITOR_MODE.HIDDEN && (
-                <Contents
-                  editJson={datafeedEditorMode === EDITOR_MODE.EDITABLE}
-                  onChange={onDatafeedChange}
-                  title={i18n.translate('xpack.ml.newJob.wizard.jsonFlyout.datafeed.title', {
-                    defaultMessage: 'Datafeed configuration JSON',
-                  })}
-                  value={datafeedConfigString}
-                />
+                <>
+                  <Contents
+                    editJson={datafeedEditorMode === EDITOR_MODE.EDITABLE}
+                    onChange={onDatafeedChange}
+                    title={i18n.translate('xpack.ml.newJob.wizard.jsonFlyout.datafeed.title', {
+                      defaultMessage: 'Datafeed configuration JSON',
+                    })}
+                    value={datafeedConfigString}
+                    heightOffset={showChangedIndicesWarning ? WARNING_CALLOUT_OFFSET : 0}
+                  />
+                  {datafeedEditorMode === EDITOR_MODE.EDITABLE && (
+                    <EuiFlexItem>
+                      <DatafeedPreview
+                        combinedJob={tempCombinedJob}
+                        heightOffset={showChangedIndicesWarning ? WARNING_CALLOUT_OFFSET : 0}
+                      />
+                    </EuiFlexItem>
+                  )}
+                </>
               )}
             </EuiFlexGroup>
+            {showChangedIndicesWarning && (
+              <>
+                <EuiSpacer />
+                <EuiCallOut
+                  color="warning"
+                  size="s"
+                  title={i18n.translate(
+                    'xpack.ml.newJob.wizard.jsonFlyout.indicesChange.calloutTitle',
+                    {
+                      defaultMessage: 'Indices have changed',
+                    }
+                  )}
+                >
+                  <FormattedMessage
+                    id="xpack.ml.newJob.wizard.jsonFlyout.indicesChange.calloutText"
+                    defaultMessage="It is not possible to alter the indices being used by the datafeed here. If you wish to use a different index, please start the job creation again select a different index pattern."
```

Comment:
nit: please start the job creation again **and** select

Reply:
I think the docs team might have a better terminology but is it `different index pattern` or `indices`?

Reply:
`indices` is correct in the first part of this message. But should it be `select a different index pattern or saved search` seeing as you can pick a saved search too?